monkey/infection_monkey/exploit/i_otp_provider.py

```diff
@@ -0,0 +1,18 @@
+import abc
+
+
+class IOTPProvider(metaclass=abc.ABCMeta):
+    """
+    IOTPProvider provides an interface for other components to get one-time passwords (OTPs).
+    Notably, this is used by exploiters during propagation to get OTPs for running new
+    Agents on exploited machines, so that they can authenticate with the Island.
```

Comment:
If we want to have this in the docstring, it should be name `IAgentOTPProvider`. That might not be a bad idea anyway.